### PR TITLE
make args testable, and test them with tox

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

**Comment by peterbe** [comment]
Perhaps I'm not caffeinated yet but what's the change here? You move the definition of the parser out of the `main` function so you can write tests? Is that what it's doing?

**Comment by graingert** [comment]
@peterbe I move the definition out of main so I can write tests, I also fix it.

**Comment by peterbe** [comment]
>  I also fix it.

What did you fix?

**Comment by peterbe** [comment]
By the way, I really think we should change it so that you can type this:
```
hashin package1==1.2.3 package2 package3=9.8.7
```
And it should assume `sha256` and the file `./requirements.txt`. If you want to change that you have to do `-r myotherreqs.txt` or `-a sha512`.

**Comment by graingert** [comment]
` "-v, --verbose", ->  '-v', '--verbose',`

**Comment by peterbe** [comment]
Can you please rebase this up so that it only has the relevant changes.

**Comment by peterbe** [comment]
I'm sorry about the mess and confusion. It seems that https://github.com/peterbe/hashin/pull/25 came in between and got landed. It's hard to be perfectly structured on side-projects.

**Comment by graingert** [comment]
Ah this still has some other handy fixes, can you reopen and I'll rebase?

**Comment by peterbe** [comment]
Awesome! You mean the `travis` and `tox` changes. Looking forward to that!
Also, can you undo and simplify your `.gitignore` changes. I really don't want that into the project.

**Comment by graingert** [comment]
@peterbe done.

**Comment by graingert** [comment]
@peterbe I didn't make any gitignore or tox changes in this PR, that commit was already merged.

File: hashin.py [hunk 1]
```diff
@@ -30,6 +31,33 @@
...
+    '-v', '--verbose',
+    help="Verbose output",
+    action="store_true",
+)
+parser.add_argument(
+    '-p', '--python-version',
+    help='Python version to add wheels for. May be used multiple times.',
+    action='append',
+    default=[],
+)
```
Comment (peterbe) [inline hashin.py]: Can this be moved into a function. Otherwise it'll run in import-time.

File: tests/test_arg_parse.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import argparse
+
+from nose.tools import eq_
+
+from hashin import parser
+
+
+def test_everything():
+    args = parser.parse_known_args(
+        ['hashin', 'reqs.txt', 'sha512', '--verbose', '--python-version', '3.5']
```
Comment (peterbe) [inline tests/test_arg_parse.py]: This ought to fail or else it's trying to hash in packages called "reqs.txt" and "sha512".

File: hashin.py (hunk 1 above)
Comment (graingert) [inline hashin.py]: that's generally when you want to run these, it just creates an object it doesn't parse any arguments.

**Comment by peterbe** [comment]
Just some linting problems. Should be easy to fix.

**Comment by graingert** [comment]
@peterbe for some reason I'm not getting travis failures for this repo.

**Comment by peterbe** [comment]
That's great! Thank you! I'll make another release.

**Comment by graingert** [comment]
@peterbe it's not worth another release for this, all it does is change tests.

**Comment by peterbe** [comment]
Actually. No release necessary. This is just refactoring on the tests.

**Comment by graingert** [comment]
this ^

**Comment by peterbe** [comment]
Right! You're a faster typer than me.

**Pull request merged**